source/extensions/grpc_credentials/file_based_metadata/config.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "envoy/extensions/grpc_credentials/file_based_metadata.pb.h"
+#include "envoy/grpc/google_grpc_creds.h"
+
+#include "common/protobuf/protobuf.h"
+
+#include "extensions/grpc_credentials/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace GrpcCredentials {
+namespace FileBasedMetadata {
+
+/**
+ * File Based Metadata implementation of Google Grpc Credentials Factory
+ * This implementation uses ssl creds for the grpc channel if available. Additionally, it uses
+ * MetadataCredentialsFromPlugin to add a static secret that is loaded from a file. The header key
```

Comment:
I wonder if you want to be able to compose these together. It seems kind of redundant to be repeating the SSL creds optional logic in every gRPC creds extensions, and maybe you want a file metadata credential factory that can be combined with some other channel creds (e.g. ALTS)? Any thoughts on how to make these factories more composable?